examples/parallel.rs

```diff
@@ -0,0 +1,351 @@
+                &self.scheme.sub_plain_inplace( &mut ct, pt); // ct := ct - pt
+                &self.scheme.rerandomize(&mut ct, &pk_);
+                ByteBuffer {
+                    buffer: ct.to_bytes()
+                }
+            })
+            .collect::<Vec<ByteBuffer>>()
+    }
+
+        // also deserializes and serializes at both ends.
+    pub fn parallel_plaintext_ciphertext_add_rerandomize(&self, ctvec: &Vec<ByteBuffer>, ptvec: &Vec<Vec<u8>>,
+    ) -> Vec<ByteBuffer> {
+        let pk_ = &self.pk;
+        let it_ct = ctvec.into_par_iter();
+        let it_pt = ptvec.into_par_iter();
+        let scheme_ = &self.scheme;
+        it_ct
+            .zip_eq(it_pt)
+            .map(|(ct_bytes, pt)| {
+                let mut ct:FVCiphertext<Scalar> = scheme_.from_bytes(&ct_bytes.buffer);
+                &self.scheme.add_plain_inplace( &mut ct, pt); // adds pt into ct
+                &self.scheme.rerandomize(&mut ct, &pk_);
+                ByteBuffer {
+                    buffer: ct.to_bytes()
+                }
+            })
+            .collect::<Vec<ByteBuffer>>()
+    }
+
+    pub fn parallel_plaintext_ciphertext_subtract(&self, ctvec: &Vec<ByteBuffer>, ptvec: &Vec<Vec<u8>>,
+    ) -> Vec<ByteBuffer> {
+        let it_ct = ctvec.into_par_iter();
+        let it_pt = ptvec.into_par_iter();
+        let scheme_ = &self.scheme;
+        it_ct
+            .zip_eq(it_pt)
+            .map(|(ct_bytes, pt)| {
+                let mut ct:FVCiphertext<Scalar> = scheme_.from_bytes(&ct_bytes.buffer);
+                &self.scheme.sub_plain_inplace( &mut ct, pt); // subtracts pt into ct
+                ByteBuffer {
+                    buffer: ct.to_bytes()
+                }
+            })
+            .collect::<Vec<ByteBuffer>>()
+    }
+
+    pub fn parallel_plaintext_ciphertext_add(&self, ctvec: &Vec<ByteBuffer>, ptvec: &Vec<Vec<u8>>,
+    ) -> Vec<ByteBuffer> {
+        let it_ct = ctvec.into_par_iter();
+        let it_pt = ptvec.into_par_iter();
+        let scheme_ = &self.scheme;
+        it_ct
+            .zip_eq(it_pt)
+            .map(|(ct_bytes, pt)| {
+                let mut ct:FVCiphertext<Scalar> = scheme_.from_bytes(&ct_bytes.buffer);
+                &self.scheme.add_plain_inplace( &mut ct, pt); // adds pt into ct
+                ByteBuffer {
+                    buffer: ct.to_bytes()
+                }
+            })
+            .collect::<Vec<ByteBuffer>>()
+    }
+
+    pub fn simple_encrypt(&self, plaintext: &FVPlaintext<u8>) -> FVCiphertext<Scalar> {
+        let scheme_ = &self.scheme;
+        let pk_ = &self.pk;
+        let ct = scheme_.encrypt(plaintext, &pk_);
+        return ct;
+    }
+
+    pub fn simple_plaintext_ciphertext_add(&self, ct:&mut FVCiphertext<Scalar>, pt: &FVPlaintext<u8>  ){
+        let scheme_ = &self.scheme;
+        scheme_.add_plain_inplace( ct, pt);
+    }
+}
+
+
+
+fn main() {
+    // Outline of PS3I HE operations using cupcake
```

Comment:
A high level question: what is the example supposed to demonstrate? If it is for parallel operations, would it be better to benchmark them?